jobs/build/olm_bundle/Jenkinsfile

```diff
@@ -0,0 +1,125 @@
+                         'Leave empty to build all (or ONLY, if defined)',
+            defaultValue: 'cluster-nfd-operator-container'
+            /* Temporarily excluding cluster-nfd-operator-container, until
+               https://github.com/openshift/cluster-nfd-operator/issues/82
+               gets solved */
+        )
+        string(
+            name: 'OLM_OPERATOR_ADVISORY',
+            description: '(Optional) Fetch OLM Operators NVRs from advisory.\n' +
+                         'Leave empty to fetch NVRs from "brew latest-build"',
+            defaultValue: ''
+        )
+        string(
+            name: 'BUNDLES_ADVISORY',
+            description: '(Optional) Attach built bundles to advisory.\n' +
+                         'Leave empty to just build without attaching them anywhere',
+            defaultValue: ''
+        )
+        booleanParam(
+            name: 'DRY_RUN',
+            description: 'Just show what would happen, without actually executing the steps',
+            defaultValue: false
+        )
+    }
+
+    stages {
+        stage('Set build info') {
+            steps {
+                script {
+                    currentBuild.displayName += " (${params.BUILD_VERSION})"
+                    currentBuild.description  = "${params.OLM_OPERATOR_ADVISORY ?: '-'} "
+                    currentBuild.description += "${params.BUNDLES_ADVISORY ?: '-'}"
+
+                    def only = olm_bundles.commonlib.cleanCommaList(params.ONLY).split(',')
+                    def exclude = olm_bundles.commonlib.cleanCommaList(params.EXCLUDE).split(',')
+                    operator_packages = (only ?: olm_bundles.get_olm_operators()) - exclude
+                }
+            }
+        }
+        stage('Get advisory builds') {
+            when {
+                expression { ! params.OLM_OPERATOR_ADVISORY.isEmpty() }
+            }
+            steps {
+                script {
+                    operator_nvrs = olm_bundles.get_builds_from_advisory().findAll {
+                        nvr -> operator_packages.any { nvr.startsWith(it) }
+                    }
+                }
+            }
+        }
+        stage('Get latest builds') {
+            when {
+                expression { params.OLM_OPERATOR_ADVISORY.isEmpty() }
+            }
+            steps {
+                script {
+                    operator_nvrs = olm_bundles.get_latest_builds(operator_packages)
+                }
+            }
+        }
+        stage('Build bundles') {
+            when {
+                expression { ! operator_nvrs.isEmpty() }
+            }
+            steps {
+                script {
+                    if (params.DRY_RUN) {
+                        bundle_packages = operator_packages.collect {
+                            it.replace('-operator', '-operator-bundle')
+                        }
+                        bundle_nvrs = olm_bundles.get_latest_builds(bundle_packages)
```

Comment:
i'm think we want per-version locking around this. otherwise it's conceivable that distgit updates could collide.

(but don't use the same lock as is being used for the non-bundle image updates)